### PR TITLE
fixes https://github.com/junegunn/vim-xmark/issues/8

### PR DESCRIPTION
**Pull request by justinkinney**
I ran into the same problem as @wookayin. For me, it was due to the fact that I had Parallels running with application sharing enabled. I'm assuming that since the windows chrome instance doesn't understand applescript, it caused the error. 

Quitting Parallels solved the problem for me, but I also updated the applescript template to include an absolute location of chrome, which fixes it even when Parallels is running.

### CODE REVIEW

File: plugin/applescript/update.scpt [hunk 1]
```diff
@@ -3,11 +3,12 @@ tell application "System Events"
...
 
 on update_xmark()
-    tell application "{{ app }}"
+    tell application "{{ app_loc }}"
         set u to "file://{{ outurl }}"
         repeat with w in every window
             set tab_index to 0
-            repeat with t in every tab of w
+            set tablist to tabs of w
+            repeat with t in tablist
```
Comment (junegunn) [inline plugin/applescript/update.scpt]: This change doesn't seem necessary and it's irrelevant to the fix.

File: plugin/xmark.vim [hunk 2]
```diff
@@ -38,6 +38,7 @@ let s:files = {
 \ 'xsize':  s:dir . '/ext/xsize'
 \ }
 let s:app = 'Google Chrome'
+let s:app_loc = '/Applications/Google Chrome.app'
```
Comment (junegunn) [inline plugin/xmark.vim]: I'm not sure, but hard-coded path makes me concerned. Is it possible on OS X to install Chrome on a different path than this? And if that's the case, and Chrome is not found there, we have no way to figure out the right path unless user provides, and we may have to fall back to `Google Chrome` as before.

File: plugin/applescript/update.scpt [hunk 3]
```diff
@@ -3,11 +3,12 @@ tell application "System Events"
 end tell
 
 on update_xmark()
-    tell application "{{ app }}"
+    tell application "{{ app_loc }}"
```
Comment (junegunn) [inline plugin/applescript/update.scpt]: `close.scpt` (which is called on `:Xmark!`) also has the code. Shouldn't we fix it as well?

File: plugin/xmark.vim (hunk 2 above)
Comment (junegunn) [inline plugin/xmark.vim]: I'm suggesting something like:
```vim
if !isdirectory(s:app_loc)
  let s:app_loc = s:app
endif
```

File: plugin/applescript/update.scpt (hunk 1 above)
Comment (justinkinney) [inline plugin/applescript/update.scpt]: This was added to eliminate the error `syntax error: Expected class name but found property. (-2741)

File: plugin/xmark.vim (hunk 2 above)
Comment (justinkinney) [inline plugin/xmark.vim]: Sounds good - I'll fix it up and update the PR.

Comment (justinkinney) [inline plugin/xmark.vim]: Also, agreed on the hard-coded path. I'm not overly familiar with applescript (or vimscript for that matter), so I'm not sure what the most reasonable approach is.

File: plugin/applescript/update.scpt (hunk 1 above)
Comment (junegunn) [inline plugin/applescript/update.scpt]: Aha, so you mean, using `s:app_loc` instead `s:app` is not sufficient to fix the problem? (sorry I can't reproduce the problem, so I'm just guessing)

Comment (justinkinney) [inline plugin/applescript/update.scpt]: For me, using the `tab` keyword seems to be the source of the syntax error, but the `tabs` keyword works. Oddly though, either seems two work. I think the key to the fix overall is the absolute path on line 6.

**Comment by justinkinney** [comment]
I'll do a bit more testing and re-submit this PR once I have a more complete understanding of what's going on.

**Comment by junegunn** [comment]
Thank you!

**Comment by justinkinney** [comment]
OK, I was able to move all of the path stuff out of the vimscript and into the close/update applescript. I had to enclose the `tell application` block in a `using terms from` block since applescript can't compile terms from the app specified using a path.

See http://daringfireball.net/2009/01/applescripts_targetting_safari_or_webkit for more detail.

File: plugin/xmark.vim [hunk 4]
```diff
@@ -103,7 +103,7 @@ function! s:xmark(resize, bang)
     execute 'augroup!' grp
     if has_key(s:tmp, bufnr('%'))
       let tmp = remove(s:tmp, bufnr('%'))
-      silent! call system(s:render('close', { 'app': s:app, 'outurl': s:urlencode(tmp) }))
+      silent! call system(s:render('close', { 'app': s:app, outurl': s:urlencode(tmp) }))
```
Comment (junegunn) [inline plugin/xmark.vim]: Missing single-quote here.
Could you squash the commits into one so that it's easier to see/track changes?

Comment (justinkinney) [inline plugin/xmark.vim]: Yep - I think it's fixed up now.

File: plugin/xmark.vim [hunk 5]
```diff
@@ -37,8 +37,8 @@ let s:files = {
 \ 'access': s:dir . '/applescript/accessibility.scpt',
 \ 'xsize':  s:dir . '/ext/xsize'
 \ }
-let s:app = 'Google Chrome'
 let s:tmp = {}
+let s:app = 'Google Chrome'
```
Comment (junegunn) [inline plugin/xmark.vim]: I'm sorry for being picky, but could you exclude this unnecessary change from the commit?

**Comment by junegunn** [comment]
Merged, thanks! :+1:
